### DISCUSSION TITLE
mini-css-extras-plugin missing "sourceMap" option

### DISCUSSION THREAD

**Discussion by NotMyFault** [webpack/webpack#16025] [General]
Hey everyone,

I'm currently in the process of updating `style-loader` from 1.2.1 to 2.0.0 in a project. The changelog recommends to update `mini-css-extract-plugin` to 1.0.0, as per https://github.com/webpack-contrib/style-loader/releases/tag/v2.0.0.
We're using 0.9.0 at the moment.

If I do that, `npx webpack --config webpack.config.js --mode=production` fails with the following exception:
```
ERROR in ./src/main/js/add-item.less (./node_modules/mini-css-extract-plugin/dist/loader.js??ref--4-1!./node_modules/css-loader/dist/cjs.js??ref--4-2!./node_modules/postcss-loader/src??ref--4-3!./node_modules/less-loader/dist/cjs.js??ref--4-4!./src/main/js/add-item.less)
Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
ValidationError: Invalid options object. Mini CSS Extract Plugin Loader has been initialized using an options object that does not match the API schema.
 - options has an unknown property 'sourceMap'. These properties are valid:
   object { publicPath?, emit?, esModule?, layer?, modules? }
    at validate (/Users/alex/idea/jenkins/war/node_modules/mini-css-extract-plugin/node_modules/schema-utils/dist/validate.js:105:11)
    at Object.pitch (/Users/alex/idea/jenkins/war/node_modules/mini-css-extract-plugin/dist/loader.js:43:29)
 @ ./src/main/js/add-item.less 2:12-294 9:17-24 13:15-22
 @ multi ./src/main/js/add-item.js ./src/main/js/add-item.less
```
The changelog doesn't outline that `sourceMap` is removed or became invalid, is there something else I need to update in order to update style-loader?
My package.json is located [here](https://github.com/jenkinsci/jenkins/blob/master/war/package.json) and the sourcemap option [here](https://github.com/jenkinsci/jenkins/blob/master/war/webpack.config.js#L70).

Thanks in advance!

**Comment by vankop** [comment]
@alexander-akait

**Reply by alexander-akait** [reply]
@NotMyFault You can enable/disable source map using:
1. the global `devtool` option
2. set [``](https://github.com/webpack-contrib/css-loader#sourcemap) to `true` so `mini-css-extract-plugin` inherit this value of `css-loader` (because not real source map generation in `mini-css-extract-plugin`)

**Reply by NotMyFault** [reply]
Thanks for the reply!

To 1., we already do that, no? https://github.com/jenkinsci/jenkins/blob/master/war/webpack.config.js#L42
To 2., what exactly can be set to true? `` seems to be empty 👀

**Reply by alexander-akait** [reply]
Fixed link https://github.com/webpack-contrib/css-loader#sourcemap (github sometimes does strange things with copy/pasting)

**Reply by alexander-akait** [reply]
But I see you already enable it, so it should work, just remove `sourceMap` option from `mini-css-extract-plugin`

**Reply by NotMyFault** [reply]
I removed `sourceMap` from the plugin configuration block https://github.com/jenkinsci/jenkins/pull/6752/commits/e7dff7bc16b21eb5ad1675d9a696a1522978f304
Local testing worked fine so far, fingers crossed 🤞🏻
